Can not build `base-test`: `/etc/apt/sources.list.d/debian.sources: No such file or directory`

**What happened:**
Unable to build the `base-test` container from `/containers` or during GL build on `macOS`.
Looks like `/etc/apt/sources.list.d/debian.sources` doesn't exist in every case or is handled in different ways by CRE (currently unable to test with `Podman`).

**Debug Logs:**
When building GL it may fail (on my `macOS`) when building the `base-test` dependency:

```
#7 0.118 sed: can't read /etc/apt/sources.list.d/debian.sources: No such file or directory
```

This file seems not to be present in every case. Full log:
```
% make
make --directory=container build-base-test
WARNING: You are still using a debian:testing-slim as a temporary replacement of gardenlinux/slim:latest!

Please run 'make slim' to fix
cp -p ../gardenlinux.asc base-test/gardenlinux.asc
mkdir -p base-test/_pipfiles
cp ../tests/Pipfile* base-test/_pipfiles
[+] Building 0.2s (8/13)                                                                                                                                                                                                                          
 => [internal] load build definition from Dockerfile                                                                                                                                                                                         0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                                                          0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                            0.0s
 => => transferring context: 2B                                                                                                                                                                                                              0.0s
 => [internal] load metadata for docker.io/gardenlinux/slim:latest                                                                                                                                                                           0.0s
 => [1/9] FROM docker.io/gardenlinux/slim                                                                                                                                                                                                    0.0s
 => [internal] load build context                                                                                                                                                                                                            0.0s
 => => transferring context: 940B                                                                                                                                                                                                            0.0s
 => CACHED [2/9] COPY gardenlinux.asc /etc/apt/trusted.gpg.d/gardenlinux.asc                                                                                                                                                                 0.0s
 => CACHED [3/9] RUN  chown root:root /etc/apt/trusted.gpg.d/gardenlinux.asc         && chmod 644 /etc/apt/trusted.gpg.d/gardenlinux.asc                                                                                                     0.0s
 => ERROR [4/9] RUN echo "deb http://deb.debian.org/debian testing main contrib non-free" > /etc/apt/sources.list      && echo "deb [signed-by=/etc/apt/trusted.gpg.d/gardenlinux.asc] http://repo.gardenlinux.io/gardenlinux today main" >  0.1s
------
 > [4/9] RUN echo "deb http://deb.debian.org/debian testing main contrib non-free" > /etc/apt/sources.list      && echo "deb [signed-by=/etc/apt/trusted.gpg.d/gardenlinux.asc] http://repo.gardenlinux.io/gardenlinux today main" >> /etc/apt/sources.list      && sed -i 's/deb.debian.org/cdn-aws.deb.debian.org/g' /etc/apt/sources.list      && sed -i 's/deb.debian.org/cdn-aws.deb.debian.org/g' /etc/apt/sources.list.d/debian.sources      && apt-get update      && apt-get install -y --no-install-recommends           curl           unzip           ca-certificates           less           apt-transport-https           gnupg           python3-pip           python3-pytest           python3-venv           vim           procps           openssh-client           inetutils-ping           wget           iproute2           libguestfs-tools           qemu-system-x86           qemu-system-arm           qemu-efi-aarch64           shellcheck           build-essential           libpython3-dev           onmetal-image:
#7 0.118 sed: can't read /etc/apt/sources.list.d/debian.sources: No such file or directory
------
executor failed running [/bin/sh -c echo "deb http://deb.debian.org/debian testing main contrib non-free" > /etc/apt/sources.list      && echo "deb [signed-by=$GARDENLINUX_MIRROR_KEY] http://repo.gardenlinux.io/gardenlinux $VERSION main" >> /etc/apt/sources.list      && sed -i 's/deb.debian.org/cdn-aws.deb.debian.org/g' /etc/apt/sources.list      && sed -i 's/deb.debian.org/cdn-aws.deb.debian.org/g' /etc/apt/sources.list.d/debian.sources      && apt-get update      && apt-get install -y --no-install-recommends           curl           unzip           ca-certificates           less           apt-transport-https           gnupg           python3-pip           python3-pytest           python3-venv           vim           procps           openssh-client           inetutils-ping           wget           iproute2           libguestfs-tools           qemu-system-x86           qemu-system-arm           qemu-efi-aarch64           shellcheck           build-essential           libpython3-dev           onmetal-image]: exit code: 2
make[1]: *** [build-base-test] Error 1
make: *** [container-test] Error 2
```

**How to reproduce it**:
* Checkout a fresh git clone
* run `export GARDENLINUX_BUILD_CRE=docker`
* run `make kvm`
 
**Environment:**
```
Darwin mac-makebox.mg0.gyptazy.ch 22.3.0 Darwin Kernel Version 22.3.0: Mon Jan 30 20:39:35 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T8103 arm64
```

| Name | Value |
| ------------------- | ----- |
| Garden Linux Branch | *main* |
| Container runtime (version)  | *Docker 4.9.1* | 
| Host OS of build machine  | *macOS* (arm64) |

--- a/container/base-test/Dockerfile
+++ b/container/base-test/Dockerfile
@@ -13,7 +13,7 @@ RUN  chown root:root $GARDENLINUX_MIRROR_KEY \
 RUN echo "deb http://deb.debian.org/debian testing main contrib non-free" > /etc/apt/sources.list \
      && echo "deb [signed-by=$GARDENLINUX_MIRROR_KEY] http://repo.gardenlinux.io/gardenlinux $VERSION main" >> /etc/apt/sources.list \
      && sed -i 's/deb.debian.org/cdn-aws.deb.debian.org/g' /etc/apt/sources.list \
-     && sed -i 's/deb.debian.org/cdn-aws.deb.debian.org/g' /etc/apt/sources.list.d/debian.sources \
+     && if [ -f "/etc/apt/sources.list.d/debian.sources"] ; then sed -i 's/deb.debian.org/cdn-aws.deb.debian.org/g' /etc/apt/sources.list.d/debian.sources ; else echo "file not present" ; fi \
      && apt-get update \
      && apt-get install -y --no-install-recommends \
           curl \

